src/main/java/randoop/condition/OutcomeTable.java

```diff
@@ -0,0 +1,99 @@
+package randoop.condition;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import randoop.test.ExpectedExceptionGenerator;
+import randoop.test.ExtendGenerator;
+import randoop.test.InvalidCheckGenerator;
+import randoop.test.PostConditionCheckGenerator;
+import randoop.test.TestCheckGenerator;
+
+/** Records the outcome of checking all of the conditions for a method. */
+public class OutcomeTable {
+  private boolean isEmpty = true;
+  private boolean hasValid = false;
+  private final List<Set<ThrowsClause>> exceptionSets;
+  private final List<PostCondition> postConditions;
+
+  public OutcomeTable() {
+    exceptionSets = new ArrayList<>();
+    postConditions = new ArrayList<>();
+  }
+
+  /**
+   * Adds the outcome of checking the conditions of a specification.
+   *
+   * @param preconditionsSatisfied boolean value indicating whether all preconditions satisfied
+   * @param throwsClauses set of exceptions expected in post-state
+   * @param postCondition post-condition that must be true in post-state, null if none
```

Comment:
They don't have to be true in the post-state in general.  They only have to be true after a normal execution, when no exception is thrown.